### DISCUSSION TITLE
How do I render .ipynb correctly?

### DISCUSSION THREAD

**Discussion by ommmirage** [quarto-dev/quarto-cli#9954] [Q&A] [answered]
### Description

Hello, everyone!

I'm creating a blog on Quarto and facing a problem of rendering .ipunb.

My notebook looks like this:
<img width="300" alt="Снимок экрана 2024-06-11 в 18 15 50" src="https://github.com/quarto-dev/quarto-cli/assets/58503543/9d0a97fa-cd67-496e-8325-5dccd1edf711">

But `quarto preview post.ipynb` shows:
<img width="400" alt="Снимок экрана 2024-06-11 в 18 17 34" src="https://github.com/quarto-dev/quarto-cli/assets/58503543/6c69502b-2baa-4fa7-b7ad-f2d118010b8f">

I tried adding metadata. Then I created new environment, reviewed my notebook and it opened just fine without any metadata. Then I return to my blog environment, tried to review my notebook outside of the blog, but it renders bad again.

quarto 1.5.43
macOS 14.5

**Comment by mcanouil** [comment]
Unfortunately, screenshots cannot be used to help you.

This being said, from the screenshot, I cannot see any metadata set, no kernels seems to be selected, and we can't see the type of the cell in which you have the text.

I suggest you take a look at the get started guide on computation (https://quarto.org/docs/get-started/computations/jupyter.html) and try to follow it.

Fore more help, you need to provide a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us  to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by ommmirage** [comment]
Github throws an error "Using nbformat v5.10.4 and nbconvert v7.16.1" to by [.ipynb file](https://github.com/ommmirage/ommmirage.github.io/blob/master/Delicious/posts/welcome/index.ipynb)

{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {

   },
   "source": [
    "Hello, everyone!\n",
    "\n",
    "My name is Sergey, I'm from Astrakhan, Russia.\n",
    "\n",
    "Last week, I started my journey into Deep Learning with the amazing teacher Jeremy Howard. His method fits me perfectly!\n",
    "\n",
    "* He is one of the best experts in the field\n",
    "* His course is modern\n",
    "* He teaches through practice\n",
    "* He simplifies as much as possible\n",
    "* I'm supposed to create a model and improve it during the course\n",
    "* We should dig deeper and deeper\n",
    "\n",
    "To succeed in DL (Deep Learning) you don't need academic background, but you need to apply what you learn from the course to your personal project and you should always be persistent.\n",
    "\n",
    "By the end of the course you'll understand all the code that's inside fast.ai, which means you've learnt the most important best practicies used in modern DL.\n",
    "\n",
    "This is my motivation, which I wrote down on my flipchart. Btw, try to write your ideas on the flipchart, maybe you'll like it as much as I do) \n",
    "\n",
    "Although Jeremy says that I should create my project right now, I'm not really interested in computer vision, which we've learned about during the first two lessons. I'm more interested in natural language processing, which we'll learn in lesson 3, and video processing.\n",
    "\n",
    "Here is the [link to his course](https://course.fast.ai/).\n",
    "\n",
    "I also have a [telegram chanel](http://https://t.me/cash900) on russian."
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  },
  "quarto": {
      "title": "Welcome To My Blog",
      "date": "2024-06-10",
      "format": "html"
    }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}

**Reply by cderv** [reply, marked as answer]
It seems that the file at https://github.com/ommmirage/ommmirage.github.io/blob/master/Delicious/posts/welcome/index.ipynb is not seen as a valid ipynb by github. 

In which tool do you created it ? Can you open it in Jupuyter lab and try to execute it ?

**Reply by ommmirage** [reply]
I changed it in Jupyter lab and now it previews correctly

**Reply by mcanouil** [reply]
The version of nbformat defined is quite old 4.2 (2016).
I would bet the issue is there.
